Plant-for-the-Planet-org/planet-webapp
# API Error handling and notifications

How to test:
- Go to undefined routes, you should see an error popup  
- After the error popup closes user will be redirected to home page


### Changes in this PR -

- [x] handle 404, 401, 403, and 500 response from the backend
- [x] show the user a message and redirect to a fallback page if necessary
- [x] validate token before making POST, PUT or DELETE requests
- [x] if /profile/* throws 404 redirect to /profile
